doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
+Set up a branch for development
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* If you will need commit/push access to the main repository,
-  `chapel-lang/chapel`_, send a request including your github username to
-  chapel_admin _at_ cray.com.
+This consists of two sets of operations.  The first (`Set up repository`_) only
+needs to happen once per developer.  The second (`Create new branch`_) should
+happen for every new effort.
 
-  .. note::
+.. _Set up repository:
 
-    You do not need commit/push access to the main repo in order to contribute
-    code. Reviewers on the core team can pull, review, and merge your pull
-    requests.
+Set up repository
++++++++++++++++++
 
-* If you're working on a long-term effort, announce it on the
-  chapel-developers_ to make sure toes are not being stepped on, work is not
-  being pursued redundantly, etc.  Similarly, fundamental changes to the
-  language or architecture should be circulated to the chapel-developers_
-  and/or chapel-users_ lists to make sure effort is not wasted.
+Note: these are expected to evolve over time as the governance of Chapel is
+migrated from Cray to an external/community body (the major elements are likely
+to be similar, though the specific people involved are likely to change and
+grow).
 
-* If your work will require committing any third-party code that you are not
-  developing yourself (or code that you've developed as a standalone package),
-  alert the chapel-developers_ mailing list of this as, presently, such code
-  packages must be approved by Cray leadership before being committed.
+#. `Set up a github account`_. The "Free" plan is sufficient for contributing to
+   Chapel.
 
-* Sign a Chapel contributor's agreement and mail it, with your GitHub
-  ID, using the instructions here:
+#. Make sure you have configured your environment to work with git. See
+   `initial git setup`_ instructions.
 
-  https://github.com/chapel-lang/chapel/tree/master/doc/rst/developer/contributorAgreements/
+#. Use the GitHub web interface to create a fork of the Chapel repo by visiting
+   https://github.com/chapel-lang/chapel and clicking the 'Fork' button (see
+   also `Fork the repo`_).  Then `configure your local git`_ and check out your
+   fork
 
-* `Fork the repo`_.
+#. If you will need commit/push access to the main repository,
+   `chapel-lang/chapel`_, send a request including your github username to
+   chapel_admin _at_ cray.com.
+
+   Note: You do not need commit/push access to the main repo in order to
+   contribute code. Reviewers on the core team can pull, review, and merge your
+   pull requests.
+
+#. If you're working on a long-term effort, announce it on the
+   chapel-developers_ mailing list to make sure toes are not being stepped on,
+   work is not being pursued redundantly, etc.  Similarly, fundamental changes
+   to the language or architecture should be circulated to the
+   chapel-developers_ and/or chapel-users_ lists to make sure effort is not
+   wasted.
```

Comment:
This is already covered under "Discuss design changes or big development efforts". I'd just remove this section. You might consider adding to "Discuss design changes or big development efforts" that it is sometimes appropriate to raise user-facing changes on chapel-users. (and if you want to describe when that is appropriate, it'd be in a a section at the bottom, if it's more than a sentence or two).